### PR TITLE
feat: reset trie updates on make_canonical

### PR DESCRIPTION
**Pull request by fgimenez**
Reset trie updates for the rest of the children forking from the canonical tip after a fork choice update.

### CODE REVIEW

**Review by rkrasiuk — APPROVED** [review]
lgtm pending @rakita

**Review by rakita — APPROVED** [review]
one nit, but lgtm

File: crates/blockchain-tree/src/blockchain_tree.rs [hunk 1]
```diff
@@ -1195,6 +1195,25 @@ where
             "Canonicalization finished"
         );
 
+        // reset trie updates for other childs
+        let chain_ids: Vec<_> = self
+            .block_indices()
+            .fork_to_child()
+            .get(&old_tip.hash)
```
[the remark this answers is not in the document]
Comment (fgimenez) [reply crates/blockchain-tree/src/blockchain_tree.rs:1202]: very good point, wasn't sure about this, thx! already removed

**Review by mattsse — APPROVED** [review]
lgtm, optional suggestion re standalone function, but didn't check if viable

**Review by mattsse — APPROVED** [review]
nice

**Pull request merged**